### DISCUSSION TITLE
Sub won't load in submarine editor

### DISCUSSION THREAD

**Discussion by Drewtron1511** [FakeFishGames/Barotrauma#13622] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I was editing my submarine and went to save changes. I opened up a different sub to look at how a component was routed. When I loaded my submarine it loaded the version I had saved 10 minutes prior. I caught back up to where I was, saved it, and the update package was selected to the wrong submarine. I selected my current submarine and saved it. I loaded it up to test to see if I fixed it and it loaded the 10min ago version again. I restarted the game. Upon loading back in, the submarine was not in the load list for the submarine editor. The submarine is in the LocalMods folder still but is not showing up in game. As of right now, my sub is lost (I have spent several hours in this one). When you go to save any sub, however, the Prototype submarine is in the save list packages.

### Reproduction steps
[Prototype Mk.I.zip](https://github.com/Regalis11/Barotrauma/files/12811237/Prototype.Mk.I.zip)
1. Add submarine file attatched into LocalMods folder
2. Start submarine editor
3. Load file of prototype (may not be present, if it is continue to step 3. If it isn't, this is bug no. 1)
4. Do some editing to prototype sub
5. Save the prototype sub and overwrite previous save.
6. Load a vanilla submarine, poke around. Don't save.
7. Load the Prototype submarine. If bug no. 2 was recreated, the overwritten save will not have taken and your changes would have been reverted.
8. Navigate to main menu and quit the game.
9. Reload game and navigate to sub editor
10. Load file. If bug was recreated, the Prototype sub will not be present.

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Drewtron1511** [comment]
Update: submarine was retrieved by loading an auto save. I renamed it and saved it. Progress was saved over top after making an edit.
[Kenny and Ely.zip](https://github.com/Regalis11/Barotrauma/files/12811270/Kenny.and.Ely.zip)
Please see the attached file for the auto save I retrieved and renamed.

**Comment by Regalis11** [comment]
Thank you for the report!

I'm glad you managed to retrieve the autosave! I gave this a quick test, attempting to follow the same steps as you, but unfortunately couldn't get the issue to occur. So I suspect it requires something more specific to happen. I wonder if it could be that you accidentally saved the sub into a different content package?

In any case, we'll keep the ticket open and see if we can figure out some way to get it to happen. Sorry for the inconvenience!

**Comment by Drewtron1511** [comment]
No problem! Me too, and happy to help.

Here is some more information if you wish to have a look.

It did end up saving over a different content package, but I don’t remember
clicking on it. I had 3 files that ended up the same submarine. All 3 files
were refusing to save any further changes until they all disappeared. It
seemed to be picking content packages to save the sub to for me, it was
confusing. I couldn’t click on different packages to overwrite and making a
new package wasn’t working either. I also remember clicking “suppress these
warnings for a session” about the shadow casting lights before the bug
occurred.

I hope this bit of information can further help! Thank you for your time! I
have been greatly enjoying your guys’ game.

**Comment by NilanthAnimosus** [comment]
Gave this a good test against 1.0.21.0 and Winter Update 1.2.4.0 both in release builds.

Couldn't reproduce the issue, I tried the steps as well as several variations outside of it (from multiple mods containing the same submarine file, saving in different mods then reloading to see what occurs, using the autosave to save, unloading all mods then saving to packages from the autosave, amongst other things)

I couldn't get it to behave at any point as described or cause them to fail to be listed or load. even when saving into an inactive mod, it would simply become active.

Hopefully its something that might have been fixed at some point, but if anyone manages to see this again more information/any other info would be appreciated.

**Comment by Drewtron1511** [comment]
I’ll be sure to let you know if I notice it again or one of my friends has
the issue! Thank you so much for looking into this so I’m depth.
Continually impressed with Barotrauma!

The issue so far hasn’t repeated, and I was able to restore the save and
continue working on that submarine! I wonder if it was just a fluke within
my own computer.

Thank you!
